src/main/java/org/opensearch/knn/plugin/transport/KNNProfileResponse.java

```diff
@@ -0,0 +1,260 @@
+ */
+
+package org.opensearch.knn.plugin.transport;
+
+import org.apache.commons.math3.stat.descriptive.AggregateSummaryStatistics;
+import org.apache.commons.math3.stat.descriptive.StatisticalSummary;
+import org.apache.commons.math3.stat.descriptive.StatisticalSummaryValues;
+import org.apache.commons.math3.stat.descriptive.SummaryStatistics;
+import org.opensearch.action.support.broadcast.BroadcastResponse;
+import org.opensearch.core.action.support.DefaultShardOperationFailedException;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.knn.profiler.SegmentProfilerState;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Response object for KNN profile requests that provides statistical information about vectors
+ * at segment, shard, and cluster levels.
+ *
+ * Example response:
+ * {
+ *   "total_shards": 2,
+ *   "successful_shards": 2,
+ *   "failed_shards": 0,
+ *   "shard_profiles": {
+ *     "0": {
+ *       "segments": [
+ *         {
+ *           "segment_id": "_0",
+ *           "dimension": 128,
+ *           "vector_statistics": [
+ *             {
+ *               "dimension_index": 0,
+ *               "statistics": {
+ *                 "count": 1000,
+ *                 "min": -0.523,
+ *                 "max": 0.785,
+ *                 "sum": 156.78,
+ *                 "mean": 0.157,
+ *                 "geometric_mean": 0.145,
+ *                 "variance": 0.089,
+ *                 "std_deviation": 0.298,
+ *                 "sum_of_squares": 245.67
+ *               }
+ *             }
+ *             // ... additional dimensions
+ *           ]
+ *         }
+ *         // ... additional segments
+ *       ],
+ *       "aggregated": {
+ *         "total_segments": 3,
+ *         "dimension": 128,
+ *         "dimensions": [
+ *           {
+ *             "dimension_id": 0,
+ *             "count": 3000,
+ *             "min": -0.723,
+ *             "max": 0.892,
+ *             "mean": 0.167,
+ *             "std_deviation": 0.312,
+ *             "sum": 501.23,
+ *             "variance": 0.097
+ *           }
+ *           // ... additional dimensions
+ *         ]
+ *       }
+ *     }
+ *     // ... additional shards
+ *   },
+ *   "cluster_aggregation": {
+ *     "total_shards": 2,
+ *     "dimension": 128,
+ *     "dimensions": [
+ *       {
+ *         "dimension_id": 0,
+ *         "count": 6000,
+ *         "min": -0.723,
+ *         "max": 0.892,
+ *         "mean": 0.172,
+ *         "std_deviation": 0.315,
+ *         "sum": 1032.45,
+ *         "variance": 0.099
+ *       }
+ *       // ... additional dimensions
+ *     ]
+ *   },
+ *   "failures": []
+ * }
+ */
+public class KNNProfileResponse extends BroadcastResponse implements ToXContentObject {
```

Comment:
nit: BroadcastResponse already implements ToXContentObject - do we need to do it again? https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/support/broadcast/BroadcastResponse.java